File: daily_problems/2025/02/0204/personal_submission/cf730a_yawn_sean.py
# Submission link: https://codeforces.com/contest/730/submission/304334332
def main():
    n = II()
    nums = LII()

    ops = []

    def op(idxs):
        for i in idxs:
            nums[i] = fmax(0, nums[i] - 1)
        res = ['0'] * n
        for i in idxs:
            res[i] = '1'
        ops.append(''.join(res))

    while min(nums) != max(nums):
        v = max(nums)
        idxs = [i for i in range(n) if nums[i] == v]
        c = len(idxs)
        
        if c > 1:
            if c % 2 == 0:
                for i in range(0, c, 2):
                    op([idxs[i], idxs[i + 1]])
            else:
                op(idxs[:3])
                for i in range(3, c, 2):
                    op([idxs[i], idxs[i + 1]])
        else:
            idx = -1
            for i in range(n):
                if nums[i] != v and (idx == -1 or nums[i] > nums[idx]):
                    idx = i
            op([idx, idxs[0]])

    print(nums[0])
    print(len(ops))
    print('\n'.join(ops))

File: daily_problems/2025/02/0204/personal_submission/cf730a_yawn_sean.cpp
#include <bits/stdc++.h>
#pragma GCC optimize("O3,Ofast,unroll-loops")
// #include "atcoder/convolution"
// #include "atcoder/dsu"
// #include "atcoder/fenwicktree"
// #include "atcoder/lazysegtree"
// #include "atcoder/math"
// #include "atcoder/maxflow"
// #include "atcoder/mincostflow"
// #include "atcoder/modint"
// #include "atcoder/scc"
// #include "atcoder/segtree"
// #include "atcoder/string"
// #include "atcoder/twosat"
// using namespace __gnu_cxx;
// using namespace __gnu_pbds;
using namespace std;
auto rng = mt19937(random_device()());
auto rngl = mt19937_64(random_device()());

// Let's set a bit and flow!
// I came, I divide, I conquer!

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(0);
    cout.tie(0);

    int n;
    cin >> n;

    vector<int> nums(n);
    for (auto &v: nums)
        cin >> v;
    
    vector<string> ops;

    auto op = [&] (vector<int> idxs) -> void {
        for (auto &i: idxs) nums[i] = max(nums[i] - 1, 0);
        string res(n, '0');
        for (auto &i: idxs) res[i] = '1';
        ops.emplace_back(res);
    };

    while (*min_element(nums.begin(), nums.end()) != *max_element(nums.begin(), nums.end())) {
        int v = *max_element(nums.begin(), nums.end());
        vector<int> idxs;
        for (int i = 0; i < n; i ++) {
            if (nums[i] == v) {
                idxs.emplace_back(i);
            }
        }
        int c = idxs.size();

        if (c > 1) {
            if (c % 2 == 0) {
                for (int i = 0; i < c; i += 2) {
                    op({idxs[i], idxs[i + 1]});
                }
            }
            else {
                op({idxs[0], idxs[1], idxs[2]});
                for (int i = 3; i < c; i += 2) {
                    op({idxs[i], idxs[i + 1]});
                }
            }
        }
        else {
            int idx = -1;
            for (int i = 0; i < n; i ++) {
                if (nums[i] != v && (idx == -1 || nums[i] > nums[idx])) {
                    idx = i;
                }
            }
            op({idx, idxs[0]});
        }
    }

    cout << nums[0] << '\n';
    cout << ops.size() << '\n';
    for (auto &x: ops) cout << x << '\n';

    return 0;
}